save_princess_2/lib/bot.rb

```diff
@@ -3,9 +3,31 @@ class Bot
   def initialize; end
 
   def nextMove(n,r,c,grid)
-    puts "0 0"
+    p_r, p_c = find_princess(grid)
+
+    if r < p_r
```

Comment:
Could consider refactoring to use a ternary operator if reducing lines was important